Lib/test/test_ctypes/test_win32_com_foreign_func.py

```diff
@@ -0,0 +1,205 @@
+import ctypes
+import gc
+import sys
+import unittest
+from ctypes import POINTER, byref, c_void_p
+from ctypes.wintypes import BYTE, DWORD, WORD
+
```

Comment:
The whole file is Windows-specific, so it would make sense to skip it here:
```suggestion

if sys.platform != "win32":
    raise unittest.SkipTest("Windows-specific test")
```
After this line you can import `HRESULT` or `ole32`, create the GUIDs, etc.

Reply:
Good catch.

As you pointed out, writing the platform bridge directly at the module level, rather than inside `setUpModule` as I did in the earlier commits, would indeed ensure that the tests are appropriately skipped on non-Windows platforms.